app/validators/date_validator.rb

```diff
@@ -52,13 +46,10 @@ def blank_fields(value)
   def blank?(value)
     return false if value.is_a?(Date)
 
-    value.to_h.slice(:day, :month, :year).all? { |_, v| v.blank? }
-  end
-
-  def empty_date_and_month?(value)
-    return false if value.is_a?(Date)
+    date_fields = %i[day month year]
+    date_fields -= [:day] if options[:month_and_year]
 
-    value.to_h.slice(:month, :year).all? { |_, v| v.blank? }
+    value.to_h.slice(*date_fields).all? { |_, v| v.blank? }
```

Comment:
Yes — so we only check for blanks in the fields we ought to have received

Reply:
Yes. Before I was attempting to do this separately for month_and_date validations which is what caused the problem as it would end up running this validation as part of the overall date checks when that did not fail.